src/tool/hpcrun/gpu/gpu-activity-process.c

```diff
@@ -455,6 +455,7 @@ gpu_kernel_block_process
 }
 
 
+__attribute__((unused))
```

Comment:
Xiaozhu has explanations in nvidia.c. I'll just revert all changes in this file.

Reply:
That would add the warning about an unused function back though, wouldn't it? I'm fine with `#if 0`'ing the whole block if that's the right approach.

Are SYNCHRONIZATION events supposed to be suppressed just for NVIDIA? Because AFAIK this is generic code, so if its a vendor-specific suppression it should be in `nvidia/*.c` somewhere. Like `cupti-activity-translate.c` with my very similar comment.

Reply:
> That would add the warning about an unused function back though, wouldn't it? I'm fine with #if 0'ing the whole block if that's the right approach.

Yes, the compiler outputs a warning message, and the warning message is also in the master branch. Synchronization events are supposed to be used for all GPUs. 

From my perspective, I like your idea to minimize code differences between two branches. So it is probably better to eliminate the differences regarding the synchronization activity in this pull request. We can create a github issue to discuss this problem and cleanup related code.